### PR TITLE
JAMES-3431 MockSMTPServer: Allow advertising extra extensions upon EHLO

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`[c3aca9f090799f25de6e178ac7738ad1e6c654e6] [ERROR] src/main/java/org/apache/james/mock/smtp/server/MockSMTPServer.java:[23,8] (imports) UnusedImports: Unused import - org.subethamail.smtp.command.EhloCommand.`

**Review by Arsnael — APPROVED** [review]

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/SMTPBehaviorRepository.java [hunk 1]
```diff
@@ -44,6 +50,26 @@ void clearBehaviors() {
         }
     }
 
+    void clearExtensions() {
+        synchronized (behaviorsInformation) {
```
Comment (LanKhuat) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/SMTPBehaviorRepository.java:54]: wrong queue?

**Comment by Arsnael** [comment]
Rebased on master

**Comment by chibenwa** [comment]
`java.lang.IllegalStateException: no known node for /home/jenkins/build/workspace/1146` test this please

**Comment by Arsnael** [comment]
```
[6e1f380b7c1dadacf539e08b7715f880c714bc73] [ERROR] Failures: 
[6e1f380b7c1dadacf539e08b7715f880c714bc73] [ERROR]   MailboxChangeTest.shouldMatchBeanContract:40 EqualsVerifier found a problem in class MailboxChange.
[6e1f380b7c1dadacf539e08b7715f880c714bc73] -> Significant fields: equals does not use isCountChange, or it is stateless.
[6e1f380b7c1dadacf539e08b7715f880c714bc73] 
[6e1f380b7c1dadacf539e08b7715f880c714bc73] For more information, go to: http://www.jqno.nl/equalsverifier/errormessages
[6e1f380b7c1dadacf539e08b7715f880c714bc73] [ERROR]   MailboxChangeTest.shouldThrowOnNullAccountId:45 
[6e1f380b7c1dadacf539e08b7715f880c714bc73] Expecting code to raise a throwable.
[6e1f380b7c1dadacf539e08b7715f880c714bc73] [ERROR]   MailboxChangeTest.shouldThrowOnNullDate:65 
[6e1f380b7c1dadacf539e08b7715f880c714bc73] Expecting code to raise a throwable.
[6e1f380b7c1dadacf539e08b7715f880c714bc73] [ERROR]   MailboxChangeTest.shouldThrowOnNullState:55 
[6e1f380b7c1dadacf539e08b7715f880c714bc73] Expecting code to raise a throwable.
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
` no known node for /home/jenkins/build/workspace/1159` test this please

**Comment by Arsnael** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**